integration/integration.test.ts

```diff
@@ -57,54 +51,70 @@ const getTests = (testType: "types-imported" | "types-in-file" = "types-in-file"
       const a3 = newAuthor({});
       expect(a3.id).toEqual("a:1");
     });
-  
+
     it("creates tagged ids based on config", () => {
       resetFactoryIds();
       const as = newAuthorSummary({});
       expect(as.id).toEqual("summary:1");
     });
-  
+
     it("fills in type name of enums", () => {
       const a = newAuthor({ popularity: { code: Popularity.Low } });
       expect(a.popularity.__typename).toEqual("PopularityDetail");
       expect(a.popularity.name).toEqual("Low");
     });
-  
+
     it("accepts codes for enum details when nested", () => {
       const a = newAuthor({ books: [{ popularity: Popularity.Low }] });
       expect(a.books[0].popularity?.name).toEqual("Low");
     });
-  
+
     it("can accept types as options with nullable references", () => {
       const book = newBook();
       const a = newAuthor({ books: [book] });
     });
-  
+
     it("can have defaults for custom scalars", () => {
       const a = newCalendarInterval();
       expect(a.start).toEqual(jan1);
       expect(a.end).toEqual(jan1);
     });
-  
+
     it("picks the 1st impl of an interface", () => {
       const c = newChild();
       expect((c.parent as any).__typename).toEqual("Author");
     });
-  
+
     it("keeps property as undefined if option is explicitly set to undefined", () => {
       const a = newAuthor({ summary: undefined });
       expect(a.__typename).toEqual("Author");
       expect(a.summary).toBeUndefined();
     });
-  
+
     it("generates property if option is not passed", () => {
       const a = newAuthor({});
       expect(a.summary.__typename).toEqual("AuthorSummary");
     });
+
+    it("cascades into children that are existing entities", () => {
+      // Given an existing factory-created instance
+      const a = newAuthor({});
+      // When create a new instance
+      const sr = newSearchResults({
+        result3: {
+          // And we destructure the existing instance
+          ...a,
+          // And override with some partial data
+          books: [{}],
+        },
+      });
+      // Then the books children have typename set
+      expect(sr.result3!.books[0].__typename).toEqual("Book");
```

Comment:
Nice! I am kind of surprised we made it this long before catching this sort of issue. I guess we didn't do much destructuring before 🤷 ?

Reply:
@bdow right. I do remember someone running into this before and saying "oh right, don't do that", but yeah not on a regular basis, which is surprising...